src/wp-admin/includes/class-wp-automatic-updater.php

```diff
@@ -486,6 +520,86 @@ public function update( $type, $item ) {
 			}
 		}
 
+		if ( 'theme' === $type ) {
+			error_log( '    Theme ' . var_export( $item->theme, true ) . ' has been upgraded.' );
+		}
+
+		if ( 'plugin' === $type ) {
+			error_log( '    Plugin ' . var_export( $item->slug, true ) . ' has been upgraded.' );
+			if ( is_plugin_inactive( $upgrader_item ) ) {
+				error_log( '    ' . var_export( $upgrader_item, true ) . ' is inactive and will not be checked for fatal errors.' );
+			}
+
+			if ( $was_active && ! is_wp_error( $upgrade_result ) ) {
+
+				/*
+				 * The usual time limit is five minutes. However, as a loopback request
+				 * is about to be performed, increase the time limit to account for this.
+				 */
+				if ( function_exists( 'set_time_limit' ) ) {
+					set_time_limit( 10 * MINUTE_IN_SECONDS );
+				}
```

Comment:
Can this call to `set_time_limit()` be moved into the `has_fatal_error()` method so it's nearer the loopback request? This is what happens in the existing `wp_edit_theme_plugin_file()` function.

Reply:
It can, although I added it here so that it would be in the context of both the loopback request and the potential rollback. It's also possible that the `::has_fatal_error()` method may be used by extenders who won't want such a long timeout to be set, though I'm not sure that's much of a concern. I don't have any strong opinions on where this is located though, so if you still think it's best placed in `::has_fatal_error()`, I'm happy to relocate it.

Reply:
I don't have a strong opinion either. Let's leave it here.